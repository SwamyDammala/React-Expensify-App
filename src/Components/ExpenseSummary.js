import React from 'react'
import { connect } from 'react-redux'
import ExpenseTotal from '../selectors/ExpenseTotal'
import SelectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpenseSummary=(props)=>{
    const expenseWord=props.expenses.length===1?'expense':'expenses'
    const formattedAmount=numeral(props.expensesTotal/100).format('$0,0.00')
    return(
            <div>
            <h1>viewing {props.expenses.length} {expenseWord} totalling {formattedAmount}</h1>
            </div>
            )    
}

// const ConnectedExpensesList=connect((state)=>{
//     return {
//         name:'swamy'
//     }
// })(ExpanseList)

const MapStateToProps=(state)=>{
    return {
        expenses:SelectExpenses(state.expenses,state.filters),
        expensesTotal:ExpenseTotal(SelectExpenses(state.expenses,state.filters))
    }
}

export default connect(MapStateToProps)(ExpenseSummary)