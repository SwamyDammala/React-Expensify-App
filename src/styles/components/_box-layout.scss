.box-layout{
    align-items: center; //it will keep the items in actual size and keep it center
  //  background: url('/images/bg.jpg');
    background-size: cover; //(It will cover the entire div)
    display: flex; //to adjest the content into exact division
    justify-content: center;  //It will make the conent to center of the window
    height:100vh; //Adjest the 100% height(vw-view height)
    width:100vw ;//Adjest the 100% width(vw-view width)
}

.box-layout__box{
    background: fade-out($color: white, $amount: 0.15);
    border-radius: 3px;
    padding:$l-size $m-size ; //distance edge of div element to content
    text-align: center;
    width:25rem;
}

.box-layout__title{
    margin:0 0 $m-size 0;
    line-height: 1;
}
