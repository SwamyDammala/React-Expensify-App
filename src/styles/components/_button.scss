.button{
    background:$blue;
    border:none;
    color:white;
    font-weight: 300;
    font-size:$font-size-large;
    padding:$s-size;
    text-decoration: none;
    display: inline-block;
    line-height: 1;
}

//For LogOut button
.button--link{
    background: none;
}

//for Remove button
.button--secondary{
    background: gray;
}