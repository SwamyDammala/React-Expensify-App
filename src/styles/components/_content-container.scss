.content-container{
    margin: 0 auto;//It will adjust the margin left and right equally
    padding: 0 $m-size;
    max-width: 80rem; //It will change the distance between total width to 80:10:10
}