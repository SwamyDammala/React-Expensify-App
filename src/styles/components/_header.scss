.header{
    background: $dark-blue;
   
}

.header__content{
    align-items: center;
    display: flex;
    justify-content: space-between;
   // padding: $s-size;
}

.header__title{
    color:white;
    text-decoration: none; //It will remove any decoratiosn on text
    h1{
        margin: 0;
    }
}