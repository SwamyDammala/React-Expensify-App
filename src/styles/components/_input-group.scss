.input-group{
    display: flex;
    margin-bottom: $l-size;
    flex-direction: column;

    @media(min-width: $desktop-breakdown){
        flex-direction: row;
        margin-bottom: $m-size;
    }
}

.input-group__items{
    margin-right:$s-size;
    @media(min-width: $desktop-breakdown){
        margin: 0 $s-size 0 0;
    }

}