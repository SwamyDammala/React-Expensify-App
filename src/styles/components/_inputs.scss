.text-input{
    border:1px solid #cdcccd;
    height:50px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;

}   

//Below is the duplicate the test input using extend

.select-choice{
    @extend .text-input
}

.textarea{
    @extend .text-input;
    height: 10rem;
}