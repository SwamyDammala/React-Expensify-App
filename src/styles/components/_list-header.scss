.list-header{
    background: #f7f7f7;
    border:1px solid darken($color: #f7f7f7, $amount: 7%);
    color:gray;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-item{
    border:1px solid darken($color: #f7f7f7, $amount: 7%);
    border-top: none;
    display: flex;
    flex-direction: column;
    color: $dark-gray;
    text-decoration: none;
    padding: $s-size;
    transition: background 0.3s ease;
    &:hover{
        background: #f7f7f7;
    }
    @media(min-width: $desktop-breakdown){
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
        align-items: center;
    }
}

.list-item--messege{
    align-items: center;
    color:gray;
    justify-content: center;
    &:hover{
        background: none;
    }
}

.list-body{
    margin-bottom: $s-size;
    @media(min-width: $desktop-breakdown) {
      margin-bottom: $l-size;  
    }
}

.list-item__title{
    margin:0;
    word-break: break-all;
}

.list-item__subtitle{
    color:gray;
    font-size: 1.4rem;
}

.list-item__data{
    margin:$s-size 0 0 0;
    @media(min-width: $desktop-breakdown) {
        margin:0;
        padding-left: $s-size;
    }
        
}





