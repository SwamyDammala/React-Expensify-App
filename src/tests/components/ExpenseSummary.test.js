import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'

const expenses=[{expensesCount:1,expensesTotal:135},{expensesCount:23,expensesTotal:23500}]

test('Should check Expense Summary for single expense',()=>{
    const wrapper=shallow(<ExpenseSummary expenses={[expenses[0]]}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should check Expense Summary for multiple expenses',()=>{
    const wrapper=shallow(<ExpenseSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})
