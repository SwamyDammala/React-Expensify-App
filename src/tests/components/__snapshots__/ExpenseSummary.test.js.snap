// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should check Expense Summary for multiple expenses 1`] = `
<div>
  <h1>
    viewing 
    2
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`Should check Expense Summary for single expense 1`] = `
<div>
  <h1>
    viewing 
    1
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;
